packages/editor-ui/src/utils.ts

```diff
@@ -62,3 +63,65 @@ export const intersection = <T>(...arrays: T[][]): T[] => {
 	const ab = a.filter(v => b.includes(v));
 	return [...new Set(rest.length ? intersection(ab, ...rest) : ab)];
 };
+
+export const isObj = (obj: unknown): obj is object => !!obj && Object.getPrototypeOf(obj) === Object.prototype;
+
+export const isSchemaTypeObjectOrList = (type: string) => ['object', 'list'].includes(type);
+
+export const getTypeof = (value: unknown): N8nJsonSchemaType => value === null
+	? 'null'
+	: value instanceof Date
+		? 'date'
+		: Array.isArray(value)
+			? 'list'
+			: typeof value;
+
+export const getJsonSchema = (input: Optional<Primitives | object>, key?: string, path = ''): N8nJsonSchema => {
+	let schema:N8nJsonSchema = { type: 'undefined', value: 'undefined' };
+	switch (typeof input) {
+		case 'object':
+			if (input === null) {
+				schema = { type: 'string', value: '[null]' };
+			} else if (input instanceof Date) {
+				schema = { type: 'date', value: input.toISOString() };
+			} else if (Array.isArray(input)) {
+				schema = {
+					type: 'list',
+					value: '',
+					path: `${path}[*]`,
+				};
+				const firstItem = input[0];
+				if(Array.isArray(firstItem)){
+					schema.value = [getJsonSchema(firstItem, '', `${path}[*]`)];
```

Comment:
how will this `[*]` work with mapping? when we map data now, we map using the index..

Reply:
That's a good question, now it suggests to me that from JSON schema view you can't map any index of a list just the list itself.
Users are not able choose an item from the list
Is this assumption true?

Reply:
You need to Max about this then

Reply:
I'll talk with Max about this but I would leave it like this here because this is the correct JSON path.
We cannot use the index, that's for sure because there is no such thing in schema (arrays are reduced to their data types, more precisely to the data types of their first item - according to the acceptance criteria)